add validation warning for exp in derived (#5731)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/derived-metrics.md

| `type` | The type of the metric (cumulative, derived, ratio, or simple). | Required |
| `label` | Required string that defines the display value in downstream tools. Accepts plain text, spaces, and quotes (such as `orders_total` or `"orders_total"`). | Required |
| `type_params` | The type parameters of the metric. | Required |

> | `expr` | The derived expression. Users will see validation warnings if the derived metric does not have an expr or if the expr does not use all the input metrics. | Required |

Reviewer: ```suggestion
| `expr` | The derived expression. You see validation warnings when the derived metric is missing an `expr` or  the `expr` does not use all the input metrics. | Required |
```
